DSCResources/MSFT_xRobocopy/MSFT_xRobocopy.psm1

```diff
@@ -157,25 +140,41 @@ function Test-TargetResource
     #  4: mismatched files/directories
     #  8: copy errors and retries exceeded
     # 16: serious error
-    if ($result -ge 0 -and $result -lt 8) {[system.boolean]$result = $true}
-    else {[system.boolean]$result = $false}
+    if ($result -ge 1 -and $result -lt 8) 
+    {
+        Write-Verbose "Source and destination are out of sync"
+        [system.boolean]$result = $false
+    }
+    elseif ($result -eq 0)
+    {
+        Write-Verbose "Source and destination are completely synchronized"
+        [system.boolean]$result = $true
+    }
+    else
+    {
+        throw "robocopy test returned with errors, exit code: $result!"
+    }
     
     $result
 }
 
 # Helper Functions
 
 function Invoke-RobocopyTest {
-param (
-    [parameter(Mandatory = $true)]
-    [System.String]
-    $source,
+    param (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $source,
 
-    [parameter(Mandatory = $true)]
-    [System.String]
-    $destination
-)
-    $output = & robocopy.exe /L $source $destination
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $destination,
+
+        [System.String]
+        $Arguments
+    )
+
+    $output = Invoke-Expression "&robocopy.exe /L `"$Source`" `"$Destination`" $Arguments"

```

Comment:
Use of Invoke-Expression is discouraged, since it might open the possibility of script injection. I think in this case you can avoid Invoke-Expresion, &, and quotes if you instead use an array for the arguments and then use splatting (@), e.g.

PS> $source = 'path with space'
PS> $target = 'pathwithoutspace'
PS> $arguments = @('/L', '/V')
PS> robocopy.exe $source $target @arguments
	...
    Source : D:\tmp\path with space\
    Dest : D:\tmp\pathwithoutspace\
    ...
    Options : *.* /V /L /DCOPY:DA /COPY:DAT /R:1000000 /W:30
    ...